app/views/images/index.html.erb

```diff
@@ -1,14 +1,17 @@
 <section class="container-fluid">
-  <% ActsAsTaggableOn::Tag.all.map(&:name).each do |tag| %>
-    <%= tag %>
+  <%= link_to("All", images_path) %>
+  <% ActsAsTaggableOn::Tag.all.map(&:name).each do |tag_name| %>
+    <%= link_to(tag_name, "?tag_name=#{tag_name}") %>
   <% end %>
 </section>
 <section class="container-fluid album">
   <div class="row justify-content-center">
-    <% @images.each do |image| %>
-      <div class="card">
-        <%= link_to image_tag(image.imageurl, class: "card-img"), image_path(image) %>
-      </div>
+    <% if @images %>
```

Comment:
If I remember correctly, if no image with such tag the `@images` will return `[]` So you don't need to do this check.